src/pubSub/test/pubSub.spec.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  publish,
+  subscribe,
+  cancelAllSubscriptions,
+  cancelSubscription,
+  cancelTopic
+} from "../pubSub";
+import waitForExpect from "wait-for-expect";
+import PubSubscriptionOptions from "../pubSubscribeOptions";
+
+waitForExpect.defaults.timeout = 500;
+waitForExpect.defaults.interval = 10;
+
+const TOPIC_NAME = "TEST";
+
+describe("Event Emitter Tests", () => {
+  beforeEach(() => {
+    cancelAllSubscriptions();
+  });
+
+  const numberData = 5000;
+  it(`should publish the message`, async () => {
+    let result;
+    subscribe(TOPIC_NAME, data => {
+      result = data;
+    });
+
+    publish(TOPIC_NAME, numberData);
+    await waitForExpect(() => expect(result).toEqual(numberData));
+  });
+
+  it(`should publish the message with multiple data parameters`, async () => {
+    let result;
+    subscribe(TOPIC_NAME, (data1, data2, data3, data4) => {
+      result = data1 + data2 + data3 + data4;
+    });
+
+    publish(TOPIC_NAME, numberData, numberData, numberData, numberData);
+
+    await waitForExpect(() => expect(result).toEqual(numberData * 4));
+  });
+
+  it(`should NOT listen to the published message`, async () => {
```

Comment:
Just FYI, you could use a [mock function](https://jestjs.io/docs/en/mock-functions) and expect it to [not be called](https://jestjs.io/docs/en/expect#tohavebeencalled).

Reply:
Great tip! Included in last commit.